diff -ru a/v2/build.go b/v2/build.go
--- a/v2/build.go	2026-10-06 11:35:33.000000000 +0200
+++ b/v2/build.go	2026-10-06 16:15:03.778397946 +0200
@@ -594,6 +594,11 @@
 		cmd.Dir = cmdDir
 		return cmd, exeExists, nil
 	case mode.Clojure:
+		if filepath.Ext(sourceFilename) == ".fnl" {
+			cmd = exec.Command("fennel", "--compile", sourceFilename)
+			cmd.Dir = sourceDir
+			return cmd, everythingIsFine, nil
+		}
 		cmd = exec.Command("lein", "uberjar")
 		projectFileExists := files.IsFile("project.clj")
 		parentProjectFileExists := files.IsFile("../project.clj")
diff -ru a/v2/run.go b/v2/run.go
--- a/v2/run.go	2026-10-06 11:35:33.000000000 +0200
+++ b/v2/run.go	2026-10-06 16:15:03.778663308 +0200
@@ -56,6 +56,15 @@
 		stopBackgroundProcesses()
 		cmd = exec.Command("chuck", sourceFilename)
 	case mode.Clojure:
+		if filepath.Ext(sourceFilename) == ".fnl" {
+			if files.Exists(filepath.Join(sourceDir, "main.lua")) && files.WhichCached("love") != "" {
+				cmd = exec.Command("love", ".")
+			} else {
+				cmd = exec.Command("fennel", sourceFilename)
+			}
+			cmd.Dir = sourceDir
+			break
+		}
 		cmd = exec.Command("clojure", "-M", sourceFilename) // single file
 	case mode.CMake:
 		cmd = exec.Command("cmake", "-B", "build", "-D", "CMAKE_BUILD_TYPE=Debug", "-S", sourceDir)
diff -ru a/v2/vendor/github.com/xyproto/mode/filename.go b/v2/vendor/github.com/xyproto/mode/filename.go
--- a/v2/vendor/github.com/xyproto/mode/filename.go	2026-10-06 11:35:33.000000000 +0200
+++ b/v2/vendor/github.com/xyproto/mode/filename.go	2026-10-06 16:15:03.776733822 +0200
@@ -125,7 +125,7 @@
 		case ".cpp", ".cc", ".c++", ".cxx", ".hh", ".hpp", ".h", ".h++": // C++ mode
 			// TODO: Find a way to discover is a .h file is most likely to be C or C++
 			mode = Cpp
-		case ".clj", ".clojure", ".cljs":
+		case ".clj", ".clojure", ".cljs", ".fnl":
 			mode = Clojure
 		case ".cs": // C#
 			mode = CS
