--- a/backends/graphics/surfacesdl/surfacesdl-graphics.cpp	2026-10-06 08:35:38.497873747 +0200
+++ b/backends/graphics/surfacesdl/surfacesdl-graphics.cpp	2026-10-06 08:35:38.517859261 +0200
@@ -2310,8 +2310,7 @@
 
 	if (keycolorChanged) {
 #if SDL_VERSION_ATLEAST(3, 0, 0)
-		uint32 flags = _disableMouseKeyColor ? 0 : SDL_SRCCOLORKEY | SDL_SRCALPHA;
-		SDL_SetSurfaceColorKey(_mouseOrigSurface, flags, _mouseKeyColor);
+		SDL_SetSurfaceColorKey(_mouseOrigSurface, !_disableMouseKeyColor, _mouseKeyColor);
 		SDL_SetSurfaceRLE(_mouseOrigSurface, !_disableMouseKeyColor);
 #else
 		uint32 flags = _disableMouseKeyColor ? 0 : SDL_RLEACCEL | SDL_SRCCOLORKEY | SDL_SRCALPHA;
@@ -2433,8 +2432,7 @@
 
 	SDL_SetColors(_mouseSurface, _cursorPaletteDisabled ? _currentPalette : _cursorPalette, 0, 256);
 #if SDL_VERSION_ATLEAST(3, 0, 0)
-	uint32 flags = _disableMouseKeyColor ? 0 : SDL_SRCCOLORKEY | SDL_SRCALPHA;
-	SDL_SetSurfaceColorKey(_mouseSurface, flags, _mouseKeyColor);
+	SDL_SetSurfaceColorKey(_mouseSurface, !_disableMouseKeyColor, _mouseKeyColor);
 	SDL_SetSurfaceRLE(_mouseSurface, !_disableMouseKeyColor);
 #else
 	uint32 flags = _disableMouseKeyColor ? 0 : SDL_RLEACCEL | SDL_SRCCOLORKEY | SDL_SRCALPHA;
