--- a/dwl.c
+++ b/dwl.c
@@ -2951,7 +2951,8 @@
 	 * Xcursor themes to source cursor images from and makes sure that cursor
 	 * images are available at all scale factors on the screen (necessary for
 	 * HiDPI support). Scaled cursors will be loaded with each output. */
-	cursor_mgr = wlr_xcursor_manager_create(NULL, 24);
+	cursor_mgr = wlr_xcursor_manager_create(cursor_theme, 24);
+	setenv("XCURSOR_THEME", cursor_theme, 1);
 	setenv("XCURSOR_SIZE", "24", 1);
 
 	/*
